### ISSUE TITLE
Nesting Observables in Vtree nodes more than one level deep

### ISSUE THREAD

**Issue by laszlokorte** [cyclejs/cycle-dom#71]
TLDR: `flatMapLatest` should be used instead of `flatMap` inside `transposeVTree` 

If an observable is nested inside a vtreenode like `vtree -> observable -> vtree -> observable -> vtree -> observable` and this nesting is more than 2 observables deep the transposition is not working correctly.

This is working:
```javascript
function main ({DOM}) {
  return {
    DOM: Observable.interval(1000).startWith(0).map((i) =>div('.hello-world', [
      'Super: ' + i,
      Observable.interval(100).startWith(0).map((j) =>div('.hello-world', [
         'Sub: ' + j, 
      ]))
    ]))
  };
}
```

Expected behavior is for the sub counter to increment once per 100ms and the super counter to increment once per second. The sub counter should reset to 1 each time the super counter increments.
This is working.

How if we increase the nesting level:

```javascript
function main ({DOM}) {
  return {
    DOM: Observable.interval(10000).startWith(0).map((i) =>div('.hello-world', [
      'Super:' + i,
      Observable.interval(1000).startWith(0).map((j) =>div('.hello-world', [
         'Sub: ' + j,
         Observable.interval(100).startWith(0).map((k) =>div('.hello-world', [
          'SubSub: ' + k,
          Observable.interval(10).startWith(0).map((h) =>div('.hello-world', 'SubSubSub ' + h))
        ]))  
      ]))
    ]))
  };
}
```
Expected would be that the SubSubSub counter increments at 100 Hz and resets each time the SubSub counter increments. The SubSub counter increments at 10 Hz and resets each time the Sub counter increments. The Sub counter increments at 1 Hz and the Super counter increments once each 10 seconds.

The actual behavior is that each time a Sub* counter changes it also resets it's parents. This causes a really weird flickering.

A real world use case for this is https://github.com/Cmdv/cycle-webpack-boilerplate/
Where the routing mechanism creates a tree like:

vtree$ -> vtreeDiv(contentArea) -> currentPage$ -> vtreeDiv(pageContainer) -> pageState$ -> vtreeDiv(pageContent)

Even deeper nestings are imaginable if a page gets composed of sub pages.

A simple fix for this is to use `flatMapLatest` instead of `flatMap` inside `transposeVTree`

**Comment by staltz** [comment]
Thanks for finding another bug.

**Comment by Frikki** [comment]
I’m taking on this issue.

**Comment by Frikki** [comment]
I find it extremely difficult to write a consistent test for this issue.

**Comment by staltz** [comment]
What have you tried so far?

**Comment by Frikki** [comment]
Updated my comment above.

**Comment by Frikki** [comment]
This issue needs to be assigned to someone else than me. After trying, I don’t know how to write a repeatable DOM render test for this issue.

**Comment by laszlokorte** [comment]
I am on it

**Comment by staltz** [comment]
Fixed in v8.0.0-rc6.
